sdk/src/Services/DynamoDBv2/Custom/DataModel/TransactGet.cs

```diff
@@ -0,0 +1,307 @@
+        /// </summary>
+        public List<object> Results => UntypedResults;
+
+        #endregion
+
+
+        #region Constructor
+
+        internal TransactGet(DynamoDBContext context, DynamoDBFlatConfig config)
+        {
+            Context = context;
+            Config = config;
+        }
+
+        #endregion
+
+
+        #region Protected methods
+
+        /// <summary>
+        /// Executes a server call to get the items requested in a transaction.
+        /// Populates Results with the retrieved items.
+        /// </summary>
+        protected internal abstract void ExecuteHelper();
+
+#if AWS_ASYNC_API
+        /// <summary>
+        /// Executes an asynchronous server call to get the items requested in a transaction.
+        /// Populates Results with the retrieved items.
+        /// </summary>
+        protected internal abstract Task ExecuteHelperAsync(CancellationToken cancellationToken);
+#endif
+        #endregion
+
+
+        #region Internal methods
+
+        internal abstract void PopulateResults();
+
+        #endregion
+    }
+
+    /// <summary>
+    /// Represents a strongly-typed object for retrieving multiple items
+    /// from a single DynamoDB table in a transaction.
+    /// </summary>
+    public class TransactGet<T> : TransactGet
+    {
+        #region Public properties
+
+        /// <summary>
+        /// List of results retrieved from DynamoDB.
+        /// Populated after Execute is called.
+        /// </summary>
+        public new List<T> Results { get { return TypedResults; } }
+
+        #endregion
+
+
+        #region Public methods
+
+        /// <summary>
+        /// Add a single item to get, identified by its hash primary key.
+        /// </summary>
+        /// <param name="hashKey">Hash key of the item to get.</param>
+        public void AddKey(object hashKey)
+        {
+            AddKey(hashKey, rangeKey: null);
+        }
+
+        /// <summary>
+        /// Add a single item to get, identified by its hash-and-range primary key.
+        /// </summary>
+        /// <param name="hashKey">Hash key of the item to get.</param>
+        /// <param name="rangeKey">Range key of the item to get.</param>
+        public void AddKey(object hashKey, object rangeKey)
+        {
+            Key key = Context.MakeKey(hashKey, rangeKey, StorageConfig, Config);
+            DocumentTransaction.AddKeyHelper(key);
+        }
+
+        /// <summary>
+        /// Add a single item to get.
+        /// </summary>
+        /// <param name="keyObject">Object key of the item to get.</param>
+        public void AddKey(T keyObject)
+        {
+            Key key = Context.MakeKey(keyObject, StorageConfig, Config);
+            DocumentTransaction.AddKeyHelper(key);
+        }
+
+        /// <summary>
+        /// Add multiple items to get.
+        /// </summary>
+        /// <param name="keyObjects">Object keys of the items to get.</param>
+        public void AddKeys(IEnumerable<T> keyObjects)
```

Comment:
I figured it'd be a nice addition (and some of the tests I've written already use it), but I agree it should be consistent between `BatchGet` and `TransactGet`. However, I'm unsure about adding other overloads. What would be their signature? I thought about the following:
```csharp
void AddKeys(IEnumerable<object> hashKeys);
void AddKeys(IEnumerable<(object, object)> hashAndRangeKeys);
```
However, this solution would cause multiple problems:
- Adding an overload that takes `IEnumerable<object>` would accept an `object`-typed enumerable with the same items as the other method which takes `IEnumerable<T>`. However, this would lead to runtime errors since the entire object would be interpreted as the hash key instead of finding its properties which represent the keys. I know this is also a problem with `AddKey`, but I think it's easier to make this mistake with enumerables than with single items. I'd prefer to distinguish the two methods by name too: the variant that takes `IEnumerable<T>` could be named `AddItems` or similar. However, that wouldn't be consistent with the existing naming convention for methods that take single items in `BatchGet`.
- `(object, object)` is a `ValueTuple` which I think is not compatible with .NET Framework 3.5, but the library is.
A new class could be created with properties for the hash key and range key, but I don't think it's worth it to create one just for to be used with these methods.
What's your opinion about this?

Reply:
Let's just keep as you have it for now. We can address `BatchGet` if we get feedback asking for a similar helper method.